enhancements/olm/olm-managed-operator-metrics.md

```diff
@@ -0,0 +1,130 @@
+
+Second, OLM will define how an operator author can:
+
+- Suggest a namespace that the operator should be deployed to
+- Specify that the operator should be scraped by the OpenShift Monitoring Prometheus Operator present on all OpenShift clusters.
+
+This information will be consumed by [OpenShift Console UI](https://github.com/openshift/console) to integrate the operator into the OpenShift Monitoring Prometheus Operator.
+
+### Goals
+
+- Provide OLM managed operators with a first class way of creating `ServiceMonitors` and `PrometheusRules` resources to interact with the Prometheus Operator
+- Enable a set of Red Hat approved Operators to integrate with the OpenShift Monitoring Prometheus Operator
+
+### Non-Goals
+
+- Provide generic metrics about OLM managed operators
+- Own the process to integrate with Telemeter
+- Support operand metric reporting
+
+## Proposal
+
+### Generic Prometheus Operator Support
+
+#### OLM support for new Prometheus objects in bundles
+
+The bundle object currently has the following format:
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   └── example.rbac.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+Notice that OLM understands how to create and manage CSVs, CRDs, and RBAC resources. This enhancement proposes adding support for Prometheus `ServiceMonitors` and `PrometheusRules` resources:
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   ├── example.prometheusrule.yaml
+    │   ├── example.rbac.yaml
+    │   └── example.servicemonitor.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+OLM would then need to be updated to identify the `ServiceMonitors` and `PrometheusRules` resources and deploy them as part of an `InstallPlan`. There is no way for OLM to identify which Prometheus Operator ServiceAccount should be granted the appropriate RBAC privileges to view the `ServiceMonitor` and `PrometheusRule` resource events in the namespace that the operator is being deployed to. As such, the cluster admin will need to configure the Prometheus Operator to watch for events in the correct namespace.
+
+### OpenShift Monitoring Prometheus Operator Support
+
+#### Requirements
+
+Once OLM offers bundle support for the `ServiceMonitor` and `PrometheusRule` resources, a sanctioned set of OLM managed operators should be able to report metrics to the OpenShift Monitoring Prometheus Operator.
+
+A number of requirements must be met for an applications metrics to be scraped by the OpenShift Monitoring Prometheus Operator.
```

Comment:
Can we mention "cluster monitoring" so we differentiate between cluster workload monitoring and user workload monitoring.